background_isolate_channels/lib/main.dart

```diff
@@ -0,0 +1,140 @@
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Background Isolate Channels',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: const MyHomePage(title: 'Background Isolate Channels'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() {
+    return _MyHomePageState();
+  }
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  /// The database that is running on a background [Isolate]. This is nullable
+  /// because acquiring a [SimpleDatabase] is an asynchronous operation. This
+  /// value is `null` until the database is initialized.
+  SimpleDatabase? _database;
+
+  /// Local cache of the query results returned by the [SimpleDatabase] for the
+  /// UI to render from. It is nullable since querying the results is
+  /// asynchronous. The value is `null` before any result has been received.
+  List<String>? _entries;
+
+  /// What is searched for in the [SimpleDatabase].
+  String _query = '';
+
+  _MyHomePageState() {
+    // Write the value to [SharedPreferences] which will get read on the
+    // [SimpleDatabase]'s isolate.
+    final Future<void> sharedPreferencesSet = SharedPreferences.getInstance()
+        .then(
+            (sharedPreferences) => sharedPreferences.setBool('isDebug', true));
+    final Future<Directory> tempDirFuture =
+        path_provider.getTemporaryDirectory();
+
+    // Wait until the [SharedPreferences] value is set and the temporary
+    // directory is received before opening the database. If
+    // [sharedPreferencesSet] does not happen before opening the
+    // [SimpleDatabase] there has to be a way to refresh
+    // [_SimpleDatabaseServer]'s [SharedPreferences] cached values.
+    Future.wait([sharedPreferencesSet, tempDirFuture])
+        .then((List<dynamic> values) {
+      final Directory? tempDir = values[1];
+      final String dbPath = path.join(tempDir!.path, 'database.db');
+      SimpleDatabase.open(dbPath).then((SimpleDatabase database) {
+        setState(() {
+          _database = database;
+          _refresh();
+        });
+      });
+    });
+  }
+
+  /// Performs a find on [SimpleDatabase] with [query] and updates the listed
+  /// contents.
+  void _refresh({String? query}) {
+    if (query != null) {
+      _query = query;
+    }
+    _database?.find(_query).toList().then((entries) {
+      setState(() {
+        _entries = entries;
+      });
+    });
+  }
+
+  /// Adds a UUID and a timestamp to the [SimpleDatabase].
+  void _addDate() {
+    final DateTime now = DateTime.now();
+    final DateFormat formatter =
+        DateFormat('EEEE MMMM d, HH:mm:ss\n${const uuid.Uuid().v4()}');
+    final String formatted = formatter.format(now);
+    _database!.addEntry(formatted).then((_) => _refresh());
```

Comment:
I was thinking a crash if someone was fast enough to hit it would be ok, but I've disabled them until the database is initialized.  Storing the commands until the database is ready would really complicate the sample.